### PR TITLE
*: add extended statistics doc and update related pages

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

- Adds a new doc named `extended-statistics.md` to align with the English version [Introduction to Extended Statistics](https://docs.pingcap.com/tidb/dev/extended-statistics)
- Updates affected pages related to statistics

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
  - https://docs.pingcap.com/tidb/dev/extended-statistics
  - https://github.com/pingcap/docs/pull/9891
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/11384

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: extended-statistics.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+```sql
+SET GLOBAL tidb_enable_extended_stats = ON;
+```
+
+该变量的默认值为 `OFF`。该系统变量的设置对所有扩展统计信息对象生效。
+
+### 添加扩展统计信息
+
+添加扩展统计信息不是一次性任务，需要为每个扩展统计信息对象都进行添加。
```
Comment (qiancai) [inline extended-statistics.md:43]:
```suggestion
添加扩展统计信息对象不是一次性任务，你需要为每个要收集的扩展统计信息分别添加对象。
```

File: extended-statistics.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+- `stats_name`：指定统计信息对象的名称，每个表的统计信息对象名称必须唯一。
+- `stats_type`：指定统计信息的类型。目前仅支持相关性 (correlation) 类型。
+- `column_name`：指定列组，可以有多个列。目前只支持指定两个列名。
+
+<details>
+<summary>实现原理</summary>
+
+为了提高访问性能，每个 TiDB 节点在系统表 `mysql.stats_extended` 中维护一份缓存，用于扩展统计信息。在添加扩展统计信息后，下次执行 `ANALYZE` 语句时，如果系统表 `mysql.stats_extended` 中有相应的对象，TiDB 将收集扩展统计信息。
+
+每一行在 `mysql.stats_extended` 系统表中都有一个 `version` 列。只要一行数据有更新，`version` 的值就会增加。这样，TiDB 会将表增量加载到内存中，而不是全量加载。
```
Comment (qiancai) [inline extended-statistics.md:63]: 这里英文里是 “Each row in the `mysql.stats_extended` table has a `version` column.”， 不确定是不是指 “`mysql.stats_extended` 系统表中的每一行都有一个 `version` 列”

File: extended-statistics.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+>
+> 不建议直接操作 `mysql.stats_extended` 系统表，否则不同 TiDB 节点上的缓存会不一致。如果误操作了该系统表，可以在每个 TiDB 节点上执行以下语句，以清除当前缓存，并重新加载 `mysql.stats_extended` 系统表：
+>
+> ```sql
+> ADMIN RELOAD STATS_EXTENDED;
+> ```
+
+</details>
+
+### 删除扩展统计信息
```
Comment (qiancai) [inline extended-statistics.md:77]:
```suggestion
### 删除扩展统计信息对象
```

File: extended-statistics.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+### 第 3 步：启用扩展统计信息
+
+将系统变量 `tidb_enable_extended_stats` 设置为 `ON`，并将 `col1` 和 `col2` 添加为扩展统计信息对象：
+
+```sql
+ALTER TABLE t ADD STATS_EXTENDED s1 correlation(col1, col2);
+```
+
+在添加完统计信息对象执行 `ANALYZE` 语句时，TiDB 会计算表 `t` 的 `col` 和 `col2` 的[皮尔逊相关系数](https://zh.wikipedia.org/zh-cn/皮尔逊积矩相关系数)，并将对象写入 `mysql.stats_extended` 系统表。
+
```
[the remark this answers is not in the document]
Comment (lilin90) [reply extended-statistics.md:142]: 中文 merge 后我再另提一个 PR 更新下英文里的几处。

**Comment by lilin90** [comment]
@qiancai Thanks for your comments! All addressed. PTAL again~

**Review by qiancai — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Review by lilin90 — COMMENTED** [review]
/approve

**Pull request merged**